rustsat/src/instances/fio.rs

```diff
@@ -66,3 +75,167 @@ pub(crate) fn open_compressed_uncompressed_write<P: AsRef<Path>>(
     }
     Ok(Box::new(io::BufWriter::new(raw_writer)))
 }
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum SolverOutput {
+    Sat(types::Assignment),
+    Unsat,
+    Unknown,
+}
+
+#[derive(Error, Debug)]
+pub enum SatSolverOutputError {
+    #[error("The output of the SAT solver is incorrect.")]
+    Nonsolution,
+    #[error("No solution line found in the output.")]
+    NoSline,
+    #[error("No value line found in the output.")]
+    NoVline,
+    #[error("Invalid solution line found in the output.")]
+    InvalidSLine,
+}
+
+//Parse a SAT solver's output
+pub fn parse_sat_solver_output<R: BufRead>(reader: R) -> anyhow::Result<SolverOutput> {
+    let mut is_sat = false;
+    let mut have_vline = false;
+    let mut solution = None;
+
+    for line in reader.lines() {
+        let line = &line?;
+
+        //Solution line
+        if line.starts_with('s') {
+            match line {
+                line if line.contains("UNSATISFIABLE") => return Ok(SolverOutput::Unsat),
+                line if line.contains("UNKNOWN") || line.contains("INDETERMINATE") => {
+                    return Ok(SolverOutput::Unknown)
+                }
+                line if line.contains("SATISFIABLE") => {
+                    /*if let Some(solution) = solution {
+                        return Ok(SolverOutput::Sat(solution));
+                    }*/
+                    is_sat = true;
+                }
+                _ => return Err(anyhow::anyhow!(SatSolverOutputError::InvalidSLine)),
+            }
+        }
+
+        //Value line
+        if line.starts_with('v') {
+            //Did I already have vline?
+            if have_vline {
+                let current_assignment: &mut Assignment = match solution {
+                    Some(ref mut assign) => assign,
+                    _ => return Err(anyhow::anyhow!(SatSolverOutputError::Nonsolution)),
+                };
+
+                current_assignment.extend_from_vline(&line)?;
+            } else {
+                solution = Some(Assignment::from_vline(&line)?);
+            }
+
+            have_vline = true;
+        }
+    }
+
+    //There is no solution line so we can not trust the output
+    if !is_sat {
+        return Err(anyhow::anyhow!(SatSolverOutputError::NoSline));
```

Comment:
```suggestion
        anyhow::bail!(SatSolverOutputError::NoSline);
```
I just learned about this macro. This makes the expression a bit simpler.